### ISSUE TITLE
Search input widget

### ISSUE THREAD

**Issue by vy0592** [flutter/flutter#6691]
Is there a search box widget that follows the search pattern in material design?

**Comment by eseidelGoogle** [comment]
Presumably you mean https://material.google.com/patterns/search.html?  Or is there a better reference?  @HansMuller would know.

**Comment by eseidelGoogle** [comment]
I also know that https://github.com/flutter/flutter/tree/master/examples/stocks has search functionality.  I don't recall if it conforms to that pattern or not however.

**Comment by vy0592** [comment]
Yes, I mean https://material.google.com/patterns/search.html.

I saw in the example they are using input widget and search icon to achieve this functionality. I also tried using input widget, but there's a problems:
The search box mentioned in the link above has solid borders. I did not find a way to hide the line in input widget so it looks a bit weird to put it inside a container with borders.

**Comment by mehmetf** [comment]
An <Input> widget that has the capability to hide its line would be sufficient.

We actually have scenarios where a magnifying glass icon is not desirable (such as when search is within the appbar). So a complete "search widget" might be an overkill.

**Comment by Hixie** [comment]
If using RawInput instead of Input is a sufficient solution to this problem, then we should add documentation to Input that points people to RawInput if they don't want the material look of an Input control.

**Comment by HansMuller** [comment]
Here's an example of using RawInput directly.

```
import 'package:flutter/material.dart';

class Home extends StatefulWidget {
  @override
  _HomeState createState() => new _HomeState();
}

class _HomeState extends State<Home> {
  GlobalKey<RawInputState> _rawInputKey = new GlobalKey<RawInputState>();
  InputValue _value = new InputValue(text: 'Hello World');

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text('Home'),
      ),
      body: new Container(
        key: _rawInputKey,
        padding: const EdgeInsets.all(16.0),
        alignment: FractionalOffset.topLeft,

        // Some simple improvements are warranted:
        // - assert if style isn't specified
        // - cursorColor should default to the text style's color
        // - if key's value is a GlobalKey then use it as the default for focusKey

        child: new RawInput(
          focusKey: _rawInputKey,
          style: Theme.of(context).textTheme.display1,
          cursorColor: const Color(0xFF00FF00),
          value: _value,
          onChanged: (InputValue newValue) {
            setState(() {
              _value = newValue;
            });
          },
          onSubmitted: (InputValue newValue) {
            print("Submitted $newValue");
            Focus.clear(context);
          },
        ),
      ),
    );
  }
}

void main() {
  runApp(new MaterialApp(home: new Home()));
}
```

**Comment by vy0592** [comment]
I was able to achieve it using the rawinput. Here's what I did.
But this also  requires me to include file 'text_selection.dart' in order to set selection control. 
And when I select a word, the word is underscored. Is there anyway to remove the underscore since the cursor is already highlighted?
I attached the screenshot at [screenshot/zOQYnsViLSg]

`var hintText = new Positioned(
        left: 0.0,
        top: 0.0,
        child: new Text('placeholder', style: hintStyle));
    if (_currentValue.text.isEmpty) {
      stack.add(hintText);
    }
    stack.add(
      new RawInput(
        key: _rawInputKey,
        platform: defaultTargetPlatform,
        focusKey: _rawInputKey,
        selectionControls: materialTextSelectionControls,
        cursorColor: theme.primaryColor,
        value: _currentValue,
        onChanged: (InputValue newValue) {
          setState(() {
            _currentValue = newValue;
          });
        },
      ),
    );`

**Comment by Hixie** [comment]
The underling of the word is part of IME. It's the word currently being typed that you can replace with autocomplete. That's not a bug.

**Comment by TitikshaDaga** [comment]
Is there a Search widget because i can't find anything about it except this link: https://material.io/guidelines/patterns/search.html#search-in-app-search ?

**Comment by mehmetf** [comment]
We have an internally developed auto-complete/search widget that will be pushed to external within a month's time frame. Stay tuned.

**Comment by mehmetf** [comment]
@DaveShuckerow FYI

**Comment by ericmand** [comment]
If this is available, I'd love a pointer. Otherwise, I'll wait patiently.

**Comment by eseidelGoogle** [comment]
I'm not sure of the status.  The bug looks like its closed. Is there more work to do here @HansMuller?

**Comment by xster** [comment]
I think the original request isn't done yet as far as I can tell. We have a universal input field but I don't think we have an out of the box search bar with all the material UI treatment and auto-complete attachments.

**Comment by xster** [comment]
@mehmetf, was the internal version released publicly already?

**Comment by mehmetf** [comment]
Not yet. We are developing in the third_party branch however the mechanism to sync this to Github is not in place yet. I am assigning this to Dave who can provide more context.

**Comment by DaveShuckerow** [comment]
Hi all,

Work is going on internal to Google on this widget; I've been resolving the approvals that we need to get to roll code written inside Google out into GitHub.

Stay tuned.

**Comment by dragonfax** [comment]
Just happen to be looking for this, and glad to see theres been an update recently.

**Comment by hundeva** [comment]
Any news? I could make use of something like this.

**Comment by ianldgs** [comment]
Just created a plugin that might help: https://pub.dartlang.org/packages/material_search

**Comment by armehta** [comment]
Any update on this ?

**Comment by BarbaricGamester** [comment]
Is the widget finally out?!

**Comment by HansMuller** [comment]
My apologies for letting this issue grow stale for SO long. Support for a search bar, per the Material Design spec, is among the goals for Flutter development this summer: https://github.com/flutter/flutter/issues/17119

**Comment by goderbauer** [comment]
Material's [Expandable Search](https://spec.googleplex.com/m2/navigation/search.html#expandable-search) is now available in Flutter as of #17629. 
For Material's [Persistent Search](https://spec.googleplex.com/m2/navigation/search.html#persistent-search) follow https://github.com/flutter/flutter/issues/17119.
